pkg/beacon/relay/gjkr/protocol.go

```diff
@@ -1655,8 +1668,49 @@ func (rm *CombiningMember) CombineGroupPublicKey() {
 	// Add reconstructed misbehaved members' individual public keys `G * z_m`.
 	for _, peerPublicKey := range rm.reconstructedIndividualPublicKeys {
 		groupPublicKey = new(bn256.G2).Add(groupPublicKey, peerPublicKey)
-
 	}
 
 	rm.groupPublicKey = groupPublicKey
 }
+
+// groupPublicKeyShares combines group public key shares for each
+// individual member in the group. Those group public key shares are
+// needed to perform the verification of relay entry signature shares coming
+// from given group member.
+func (rm *CombiningMember) groupPublicKeyShares() map[group.MemberIndex]*bn256.G2 {
+	groupPublicKeyShares := make(map[group.MemberIndex]*bn256.G2)
+
+	// Calculate group public key shares for all other operating members.
+	for _, receiverID := range rm.group.OperatingMemberIDs() {
+		if receiverID == rm.ID {
+			continue
+		}
+
+		// Calculate first public key share for given receiver based on
+		// current member public key share points.
+		sum := rm.publicKeyShare(receiverID, rm.publicKeySharePoints)
+
+		// Iterate through the `QUAL` set and calculate subsequent
+		// public key share for given receiver based on...
+		for senderID := range rm.receivedQualifiedSharesS {
+			// ...received and valid other members public key share points...
```

Comment:
I'd say `...received and valid member's public key share points` - I think it better explains what are we doing here.